Src/contrib/library/web/authentication/oauth/cypress/consumer/src/oauth/oauth_20_jwt_service.e

```diff
@@ -0,0 +1,88 @@
+note
+	description: "Summary description for {OAUTH_20_JWT_SERVICE}."
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	OAUTH_20_JWT_SERVICE
+
+inherit
+
+	OAUTH_20_SERVICE
+		redefine
+			access_token_post,
+			config
+		end
+create
+	make
+
+feature -- Status Report
+
+	is_client_secret_required: BOOLEAN
+			-- Is client secret required?, by default false.
+
+	is_client_id_required: BOOLEAN
+			-- Is client secret required?, by default false.
+
+feature -- Change Element
+
+	enable_client_secret
+		do
+			is_client_secret_required := True
+		ensure
+			is_client_secret_required_true: is_client_secret_required = True
+		end
+
+	enable_client_id
+		do
+			is_client_id_required := True
+		ensure
+			is_client_id_required_true: is_client_id_required = True
+		end
+
+feature -- Access
+
+	access_token_post (a_request_token: detachable OAUTH_TOKEN; a_verifier: detachable OAUTH_VERIFIER): detachable OAUTH_TOKEN
+			-- retrieve an access token using a request token
+			-- (obtained previously)
+		local
+			l_request: OAUTH_REQUEST
+		do
+			create l_request.make (api.access_token_verb, api.access_token_endpoint)
+			if attached config.grant_type as l_grant_type then
+				l_request.add_body_parameter ({OAUTH_CONSTANTS}.grant_type, l_grant_type)
+			end
+			if is_client_id_required and then not config.api_key.is_empty then
```

Comment:
is_empty is fine, but I tends to use  `is_whitespace` to exclude empty case, and blank text as well.